pkg/logql/rangemapper_test.go

```diff
@@ -0,0 +1,1138 @@
+package logql
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_SplitRangeInterval(t *testing.T) {
+	rvm, err := NewRangeMapper(2 * time.Second)
+	require.NoError(t, err)
+
+	for _, tc := range []struct {
+		expr     string
+		expected string
+	}{
+		{
+			`bytes_over_time({app="foo"}[3s])`,
+			`sum without(
+				downstream<bytes_over_time({app="foo"}[1s] offset 2s), shard=<nil>>
+				++ downstream<bytes_over_time({app="foo"}[2s]), shard=<nil>>
+			)`,
+		},
+		{
+			`count_over_time({app="foo"}[5s])`,
+			`sum without(
+				downstream<count_over_time({app="foo"}[1s] offset 4s), shard=<nil>>
+				++ downstream<count_over_time({app="foo"}[2s] offset 2s), shard=<nil>>
+				++ downstream<count_over_time({app="foo"}[2s]), shard=<nil>>
+			)`,
+		},
+		{
+			`rate({app="foo"}[4s] offset 1m)`,
+			`(sum without(
+				downstream<count_over_time({app="foo"}[2s] offset 1m2s), shard=<nil>>
+				++ downstream<count_over_time({app="foo"}[2s] offset 1m0s), shard=<nil>>
+			) / 4)`,
+		},
+	} {
+		tc := tc
+		t.Run(tc.expr, func(t *testing.T) {
+			t.Parallel()
+			noop, mappedExpr, err := rvm.Parse(tc.expr)
+			require.NoError(t, err)
+			require.Equal(t, removeWhiteSpace(tc.expected), removeWhiteSpace(mappedExpr.String()))
+			require.Equal(t, false, noop)
+		})
+	}
+}
+
+func Test_SplitRangeVectorMapping(t *testing.T) {
+	rvm, err := NewRangeMapper(time.Minute)
+	require.NoError(t, err)
+
+	for _, tc := range []struct {
+		expr       string
+		expected   string
+		expectNoop bool
```

Comment:
nit: this test is massive. I think we could move all the noop cases to their own test for ease of finding them / organizational sake, which means we could get rid of this property in the struct. WDYT?